File: MyHordesOptimizerApi/MyHordesOptimizerApi/Dtos/MyHordesOptimizer/CitizenState/CitizenStateDto.cs
using System.Collections.Generic;

namespace MyHordesOptimizerApi.Dtos.MyHordesOptimizer.CitizenState
{
    public class CitizenStateDto
    {
        public int Ap { get; set; }
        public int Sp { get; set; }
        public bool Wounded { get; set; }
        public bool IsEclaireur { get; set; }
        public bool HasBike { get; set; }
        public bool HasShoes { get; set; }
        public int WalkingDistance { get; set; }
        public bool IsDead { get; set; }
        public List<string> Statuses { get; set; } = new();
        public bool HasShield { get; set; }
        public bool HasDefenceCpItem { get; set; }
        public bool IsGuide { get; set; }
        public int ZoneCitizenCount { get; set; }
        public bool HasCleanPdcPerk { get; set; }
        public bool HasHydratedPdcPerk { get; set; }
        public bool HasSoberPdcPerk { get; set; }
        public bool HasBaseZoneControlPerk { get; set; }
        public bool IsRoleGhoul { get; set; }
        /// <summary>Calculé côté serveur (CitizenPdcRules) — toute valeur reçue en entrée est ignorée.</summary>
        public int Pdc { get; set; }
    }
}

File: MyHordesOptimizerWebsite/src/app/_abstract_model/dto/citizen-state.dto.ts
export interface CitizenStateDTO {
    ap: number;
    sp: number;
    wounded: boolean;
    isEclaireur: boolean;
    hasBike: boolean;
    hasShoes: boolean;
    walkingDistance: number;
    isDead: boolean;
    statuses: string[];
    hasShield?: boolean;
    hasDefenceCpItem?: boolean;
    isGuide?: boolean;
    zoneCitizenCount?: number;
    hasCleanPdcPerk?: boolean;
    hasHydratedPdcPerk?: boolean;
    hasSoberPdcPerk?: boolean;
    hasBaseZoneControlPerk?: boolean;
    isRoleGhoul?: boolean;
    /** Calculé côté serveur — toujours présent en sortie, ignoré si fourni en entrée. */
    pdc?: number;
}
